File: 03-Array-Linkedlist/Array/283-move_zeros/solution3.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        //　双指针法,标注最后一个不为零元素的位置
        int lastNonZero = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != 0) {
                nums[lastNonZero++] = nums[i];
            }
        }

        for(int j = lastNonZero; j < nums.size(); j++) {
            nums[j] = 0;
        }
    }
};

File: 03-Search/11-BinarySearch/typeA-search-ele-idx/81-search-in-rotated-sorted-array-ii/solution3.java
 // 玄学解法：不保证结果对错


class Solution {
    public boolean search(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        
        while (left <= right) {
            int mid = left + (right - left) / 2;
            // 二分时，把中点、两个端点一块检查
            if (nums[mid] == target || nums[left] == target || nums[right] == target) {
                return true;
            } else if (nums[left] < target && target < nums[mid]) {
                right = mid;
            } else if (nums[mid] < target && target < nums[right]) {
                left = mid;
            }
            left++;
            right--;
        } 
        return false;
    }
}